test/raft/raft_address_map_test.cc

```diff
@@ -135,40 +116,31 @@ SEASTAR_THREAD_TEST_CASE(test_raft_address_map_operations) {
         BOOST_CHECK(!m.find(id2));
     }
     {
-        // Throw on re-mapping address for the same id
-        scoped_no_abort_on_internal_error abort_guard;
-        sharded<raft_address_map<manual_clock>> m_svc;
+        // Do not throw on re-mapping address for the same id
+        // - happens when IP address changes after a node restart
+        sharded<raft_address_map_t<manual_clock>> m_svc;
         m_svc.start().get();
         auto stop_map = defer([&m_svc] { m_svc.stop().get(); });
         auto& m = m_svc.local();
 
-        m.set(id1, addr1, false);
-        BOOST_CHECK_THROW(m.set(id1, addr2, false), std::runtime_error);
+        m.add_or_update_entry(id1, addr1);
+        m.set_nonexpiring(id1);
+        BOOST_CHECK_NO_THROW(m.add_or_update_entry(id1, addr2));
     }
     {
-        // Check that transition from regular to expiring entry is not possible
-        sharded<raft_address_map<manual_clock>> m_svc;
+        // Check that add_or_update_entry() doesn't transition the entry type
+        // to expiring.
+        sharded<raft_address_map_t<manual_clock>> m_svc;
         m_svc.start().get();
         auto stop_map = defer([&m_svc] { m_svc.stop().get(); });
         auto& m = m_svc.local();
 
-        m.set(id1, addr1, false);
-        m.set(id1, addr1, true);
+        m.add_or_update_entry(id1, addr1);
+        m.set_nonexpiring(id1);
+        m.add_or_update_entry(id1, addr2);
         manual_clock::advance(expiration_time);
         BOOST_CHECK(m.find(id1));
```

Comment:
Could also check if the value is correct.

Reply:
Done.